### DISCUSSION TITLE
What is the recommended way (field) to Search photos?

### DISCUSSION THREAD

**Discussion by ziggysid** [photoprism/photoprism#4645] [Q&A]
On the face of it there are three fields that could be used for searching: Subject, Keywords and Labels. But I've found they are not equal, and I'm wondering; if I'm going to invest my time to improve the data for searching, which one of these fields is best? And for background, my photos are all family photos, holidays, friends & family, weddings, parties, pets, etc.

Subject seems to be auto filled from the meta data. This gives me a good transfer of data from the photo file to Photoprism, but doesn't seem to be searchable? You would have expected that to be an easy fix.
Keywords seems to be AI populated, and can be very random with the words added here. It is searchable, but only as good as the "random" data that has been entered. By way of example of random; I took a photo of a red car in the UK, when indexing photoprism added the keywords "south atlantic ocean". Where did that come from?
Labels, very sparsely populated but is searchable. Its like a categorisation field, needs some planning to be consistent across all your photos. Could be quite onerous.

**Comment by omerdduran** [comment]
Hi. Subject, keywords, and labels serve different purposes in PhotoPrism:

Subject is specifically for searching people tagged in photos. Use `subject:"John Doe"` to find photos of specific people.

Keywords contain a list of search terms extracted from metadata, file names and other sources such as geodata. Images with matching keywords are automatically shown in the corresponding labels.

Although related, keywords and tags serve different purposes:
Labels can have main categories and are primarily used for categorization, such as "animal", "cat" or "boat". Duplication and ambiguity should be avoided. 
Keywords are primarily used for searching. They can include similar terms and translations such as 'kitten', 'kitty' and 'cat'.

Also for strange AI classifications (such as your "South Atlantic" example) you can:
   - Remove incorrect keywords in the photo editing dialog
   - Add your own relevant keywords

For more information:
https://docs.photoprism.app/user-guide/search/filters/#using-search-filters
https://docs.photoprism.app/user-guide/organize/people/#search

**Reply by ziggysid** [reply]
I've also noticed the Keywords field doesn't like numbers. I hear you saying "the clue is in the name (keyWORDS)", but to me "21st Birthday" seem like pretty good Keywords. However, PhotoPrism deletes the numbers just leaving me with "st Birthday". Frustrating!

**Reply by graciousgrey** [reply]
There will be some first improvements related to this in our next release. 

- https://github.com/photoprism/photoprism/issues/3447
